src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/DefaultHttpClient.java

```diff
@@ -8,26 +8,22 @@
 import java.util.Collection;
 import java.util.List;
 import java.util.Map;
+import java.util.concurrent.TimeUnit;
 import java.util.concurrent.locks.Lock;
 import java.util.concurrent.locks.ReentrantLock;
 
 import io.reactivex.Single;
 import io.reactivex.subjects.SingleSubject;
-import okhttp3.Call;
-import okhttp3.Callback;
-import okhttp3.Cookie;
-import okhttp3.CookieJar;
-import okhttp3.HttpUrl;
-import okhttp3.OkHttpClient;
-import okhttp3.Request;
-import okhttp3.RequestBody;
-import okhttp3.Response;
-import okhttp3.ResponseBody;
+import okhttp3.*;
 
 final class DefaultHttpClient extends HttpClient {
     private final OkHttpClient client;
 
     public DefaultHttpClient() {
+        this(5000);
+    }
+
+    public DefaultHttpClient(int pollTimeout) {
```

Comment:
Should we introduce a client options object in case we add more later?

Reply:
I'd also suggest using a TimeSpan like type so units don't get mixed up.

Reply:
We initially used the Duration class but removed it later when we found that it required Android API level 26 or later.

Reply:
We can change it to use the `TimeUnit` pattern, or change the argument to "timeoutInMilliseconds".
Speaking of which, don't call it "poll" anything because this is not specific to longpolling even if it's only ever used by longpolling.